kubernetes-client/src/test/java/io/fabric8/kubernetes/client/dsl/internal/CustomResourceOperationsImplTest.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.kubernetes.client.dsl.internal;
+
+import java.io.IOException;
+
+import io.fabric8.kubernetes.api.model.KubernetesResource;
+import io.fabric8.kubernetes.api.model.apiextensions.CustomResourceDefinition;
+import io.fabric8.kubernetes.api.model.apiextensions.CustomResourceDefinitionBuilder;
+import io.fabric8.kubernetes.client.CustomResource;
+import io.fabric8.kubernetes.client.CustomResourceList;
+import io.fabric8.kubernetes.internal.KubernetesDeserializer;
+
+import org.junit.jupiter.api.Test;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.MappingJsonFactory;
+
+import static org.hamcrest.core.IsInstanceOf.instanceOf;
+import static org.junit.Assert.assertThat;
+
+public class CustomResourceOperationsImplTest {
+
+	public static class MyCustomResource extends CustomResource {
+    }
+
+	public static class MyCustomResourceList extends CustomResourceList<MyCustomResource> {
+    }
+
+    CustomResourceDefinition crd = new CustomResourceDefinitionBuilder()
```

Comment:
Please add access modifiers to these field (private)

Ideally field instantiation should happen inside a test setup method (`@BeforeEach`) in order to avoid side effects when new tests are added to the suite.

Reply:
Changed to `private final` this should cause no side effects that I can think of.

Reply:
Making it final doesn't make it immutable which may lead to some side-effects:

So if one of the test cases (not at the moment, and probably not ever) does something like `crd.setApiVersion("bar")` in order to do some corner-case test and some other test asserts on this field, it will fail. It will also fail in a non-deterministic way depending on how the system computes the test execution order.

That's why it's always recommended to use setup and tear down methods in tests.